### PR TITLE
improve software categories

### PR DESCRIPTION
**Pull request by fembau**
- add default categories #992 
- add provenance to link to controlled vocabularies #1141 

### How to test:

see #984

### CODE REVIEW

**Comment by fembau** [comment]
Implementation of [table representation](https://github.com/research-software-directory/RSD-as-a-service/issues/992#issuecomment-2138904117) is moved into another PR to speed up community feature PR

**Comment by dmijatovic** [comment]
@fembau To fix failing unit test please add empty categories array to `frontend/__tests__/__mocks__/softwareIndexData.js`

```javascript
const softwarePageData = {
  slug: 'test-slug',
  software: softwareItem,
  citationInfo: SoftwareCitationItems,
  keywords: apiKeywordsBySoftware,
  licenseInfo,
  repositoryInfo: repoInfo,
  mentions: apiMentions,
  testimonials: apiTestimonial,
  contributors: apiContributors,
  relatedTools: apiRelatedSoftware,
  relatedProjects: apiRelatedProjects,
  isMaintainer: false,
  organisations: apiOrganisationsOfSoftware,
 // add categories to mock data
  categories:[]
}
```

**Review by cmeessen — APPROVED** [review]
Great work, thank you :+1:

File: frontend/components/software/edit/links/AutosaveSoftwareCategories.tsx [hunk 1]
```diff
@@ -1,33 +1,45 @@
-// SPDX-FileCopyrightText: 2023 Felix Mühlbauer (GFZ) <felix.muehlbauer@gfz-potsdam.de>
-// SPDX-FileCopyrightText: 2023 Helmholtz Centre Potsdam - GFZ German Research Centre for Geosciences
+// SPDX-FileCopyrightText: 2023 - 2024 Felix Mühlbauer (GFZ) <felix.muehlbauer@gfz-potsdam.de>
+// SPDX-FileCopyrightText: 2023 - 2024 Helmholtz Centre Potsdam - GFZ German Research Centre for Geosciences
 // SPDX-FileCopyrightText: 2024 Dusan Mijatovic (Netherlands eScience Center)
 // SPDX-FileCopyrightText: 2024 Netherlands eScience Center
 //
 // SPDX-License-Identifier: Apache-2.0
 
 import {ChangeEventHandler, Fragment, useEffect, useMemo, useState} from 'react'
```
Comment (dmijatovic) [inline frontend/components/software/edit/links/AutosaveSoftwareCategories.tsx:8]: - Remove unused import of useEffect at line 8
- Remove unused import of genCategoryTree at line 10
- Remove unused import of getAvailableCategories at line 11

Based on sonarcloud feedback https://sonarcloud.io/project/issues?id=rsd-frontend&pullRequest=1143&resolved=false&sinceLeakPeriod=true

File: frontend/components/software/CategoriesSection.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import {CategoryTreeLevel} from '../category/CategoryTree'
+
+export type CategoriesSectionProps = {
+  categories: CategoryPath[]
+}
+export default function CategoriesSection({categories}: CategoriesSectionProps) {
+  const tree = useCategoryTree(categories)
+
+  return tree.map((level, index) => (
+    <PageContainer className="py-12 px-4 lg:grid lg:grid-cols-[1fr,4fr]" key={index}>
```
Comment (dmijatovic) [inline frontend/components/software/CategoriesSection.tsx:18]: If possible, use unique value from data. Maybe `level.category.name` or `level.category.id` can be used as unique key? React use key value to determine data changes and re-render changed items. If the key is not unique than the changes will not be reflected (shown on the screen).

File: frontend/components/software/edit/links/SoftwareLinksInfo.tsx [hunk 3]
```diff
@@ -30,6 +41,30 @@ export default function SoftwareLinksInfo() {
       <p className="py-2"><strong>Keywords</strong></p>
       <p>Here you can provide keyword that describe your software, by selecting from existing keyword, adding your own, or importing the keywords from the Software DOI.</p>
 
+      {helpForCategories.map(([headline, text], index) => (
+        <Fragment key={index}>
```
Comment (dmijatovic) [inline frontend/components/software/edit/links/SoftwareLinksInfo.tsx:45]: Similar suggestion for key. Maybe better to use headline if it is unique?

**Review by dmijatovic — APPROVED** [review]
Nice work! I placed few minor suggestions based on sonar cloud advice.

File: frontend/components/software/CategoriesSection.tsx (hunk 2 above)
Comment (fembau) [reply frontend/components/software/CategoriesSection.tsx:18]: I know how React handles lists. Here the list does not change ;) But if the linter complains it should be fixed, of course :)

**Pull request merged**